### ISSUE TITLE
HTTPS

### ISSUE THREAD

**Issue by C9Glax** [C9Glax/tranga#4]
API is currently running only on 6531 via http.
When using reverse proxies with ssl, site doesn't work.

**Comment by C9Glax** [comment]
The problem is, that tranga-api is using http (non-s) and browsers will block http requests from https sites (as that would make the whole point of https pointless).
So until I get the will to wrap my head around SSL-certs and a proper implementation, this will continue to occur.

I think there is a hacky way you could get it running with https, but honestly since tranga should be a local-only program, http is all you (should) really need.

**Comment by JohnnyBigoud** [comment]
I tend to agree. This is just about being able to manage task list from remote location when unconnected to lan.

Easy workaround is using personal VPN though.

**Comment by voidwalkerb** [comment]
I was able to put Tranga behind Nginx Proxy Manager by creating an entry for both the web frontend and the API.

**Comment by C9Glax** [comment]
So basically reverse-proxying the api as well and then just pointing the website at that proxy?

**Comment by voidwalkerb** [comment]
Exactly though the API URL seems to need set for each browser instance. I had to set it for each device I tested it with.

**Comment by C9Glax** [comment]
> Exactly though the API URL seems to need set for each browser instance

Yea that is something that is currently not very convenient, it's stored in a cookie local to the machine.

**Comment by m8dotpie** [comment]
May be this my comment will be of any help: [link](https://github.com/C9Glax/tranga/issues/55#issuecomment-1872915473)

**Comment by db-2001** [comment]
@C9Glax I don't have it set up to SSL so I can't test, but in the most recent PR, I changed the API-URI in the JS code to use the protocol of the browser instead of defaulting to HTTP, might have fixed the issue, might not have.

**Comment by dare2try** [comment]
While not exactly related to HTTPS, this issue seems to relate well.  As with @db-2001 I was attempting to run on HTTPS via a reverse proxy (Traefik to be exact)  E.g.
```
                                                 -> web
                                                /
https://tranga.{domain}.{tld} -> reverse proxy +
                                                \
                                                 -> api
```
The hiccup in this was that I could not override the port number in the `apiConnector.js` without mounting a completely different version into the container, which I did, and instead routed the API calls to `/api` without the port number.  Unfortunately that caused other issues since the APIs aren't hosted under a path of `/api` which means I need to strip it off before sending to the actual API.  Anyway, things started to get complicated and I stopped to do other things.

Personally, I think the service setup is fine, even if I have to override a `*.js` file (a `settings.js` file would be nice instead of the actual connector js), but would it be possible to run the API under the root path of  `/api` that way integrating the two is easier?

**Comment by dare2try** [comment]
@C9Glax, once I get it all working I can always provide my own traefik config, if you want to check it out and/or use it for documentation

**Comment by db-2001** [comment]
@dare2try reverse proxying the /api should strip it automatically, no?

**Comment by dare2try** [comment]
No... the reverse proxy only changes the host unless you add in additional logic before it forwards it to the upstream service.  In my case I add `/api` to the `apiConnnector.js`, then have a routing rule in traefik that says `Host("tranga.{domain}.{tld}" && PathPrefix("/api/")`.  Leaving it as is will change the host address but leave the path untouched, so I have to remove the `/api` using a middleware called `StripPrefix` where I remove the `/api` piece.  I've got everything working now, but if you want to make it easier for others to integrate the two I was recommending you might want to add the `/api` (or something you prefer) on your API routes.

**Comment by db-2001** [comment]
Resolved by C9Glax/tranga-website#69

**Comment by dare2try** [comment]
> Make sure youre looking at the code for the `dev` branch. I had made that change in an earlier commit that is separate from the PR

Your right, my bad 👍
